lib/Models/WebMapServiceCatalogItem.js

```diff
@@ -1143,6 +1156,40 @@ WebMapServiceCatalogItem.prototype.exportData = function() {
     callWebCoverageService(this);
 };
 
+WebMapServiceCatalogItem.prototype.searchDataForMatchingFeatures = function(searchText, regex) {
+    const that = this;
+    return new Promise(function(resolve, reject) {
+        let workbenchItemResults = [];
+        if (that.hasLinkedWfs === false) {
+           resolve(workbenchItemResults);
+        } else if (that.hasLinkedWfs) {
+            that.linkedWfs.searchDataForMatchingFeatures(searchText, regex).then(function (res) {
+                workbenchItemResults = convertWfsSearchResultsToWms(res, that);
+                resolve(workbenchItemResults);
+            });
+        } else if (!defined(that.hasLinkedWfs)) {
+            loadXML(that.getCapabilitiesUrl.replace(/wms/gi, 'wfs'))
+            .then(function(xml) {
+                that.linkedWfs = new WebFeatureServiceCatalogItem(that.terria);
```

Comment:
We work with a lot of WMS servers that have so much data that loading the whole thing with a single WFS will crash the server, the browser tab, or both. Instead, we should construct a WFS `GetFeature` query to do the search on the server.